GoodNightProject/GoodNightProject/Views/AboutPage.xaml.cs

```diff
@@ -183,46 +203,122 @@ public AboutPage()
             string json = Preferences.Get("times", "[]");
             times = Newtonsoft.Json.JsonConvert.DeserializeObject<List<TimeSpan>>(json);
+            dayWhenAlarmGoes = DateTime.Parse(Preferences.Get("dayWhenAlarmGoes","1.01.2023"));
+            string json2 = Preferences.Get("recommendationList", "[]");
+            recommendationList = Newtonsoft.Json.JsonConvert.DeserializeObject<List<bool>>(json2);
+    
             if (isAlarmSet == true)
             {
                 time.IsEnabled = false;
             }
             firstTimeAppUsing = bool.Parse(Preferences.Get("firstTimeAppUsing", "true"));
+            rateAlarm = bool.Parse(Preferences.Get("rateAlarm", "true"));
+
+            if(aboutPageSetPrefereces)
+            {
+                if(rateAlarm && dayWhenAlarmGoes < DateTime.Now) 
+                {
+                    Recomandation();
+                }
+                aboutPageSetPrefereces = false;
+            }
+            Timer();
+        }
+        private void Timer()
+        {
+            if (isAlarmSet)
+            {
+                Device.StartTimer(TimeSpan.FromSeconds(1), () =>
+                {
+                    if (DateTime.Now >= dayWhenAlarmGoes)
+                    {
+                        Recomandation();
+                    }
+                    return true;
+                });
+            }
+        }
+        private void Recomandation()
+        {
+            var x = DisplayAlert("Alarm", "Czy wyspales sie?", "Tak", "Nie");
+            if (x.Result)
+            {
+                recommendationList.Add(true);
+            }
+            else
+            {
+                recommendationList.Add(false);
+            }
+            Preferences.Set("recommendationList", Newtonsoft.Json.JsonConvert.SerializeObject(recommendationList));
+            dayWhenAlarmGoes = dayWhenAlarmGoes.AddDays(1);
+            if(isAlarmSet)
+            {
+                IAlarmService alarmService = DependencyService.Get<IAlarmService>();
+                var godzina = algorithm(selectedTime.Hours, selectedTime.Minutes);
+                alarmService.CancelAlarm();
+                alarmService.CancelMedia();
+                alarmService.SetAlarm(godzina.Item1, godzina.Item2);
+            }
         }
         public (int, int) algorithm(int hour, int minute)// Algorytm do ustawiania alarmu na najbliższą 1,5 minuty
         {
             DateTime teraz = DateTime.Now; // Pobierz aktualną godzinę i minutę
 
-            DateTime podanaData = new DateTime(teraz.Year, teraz.Month, teraz.Day, hour, minute, 0); // RObie obiekt DateTime dla podanej godziny i minuty
+            DateTime podanaData = new DateTime(teraz.Year, teraz.Month, teraz.Day, hour, minute, 0); // Obiekt DateTime dla podanej daty
 
-            if(podanaData < teraz) // Jeżeli podana godzina jest mniejsza od teraźniejszej godziny
+            teraz = teraz.AddMinutes(25); //dodaje czas zasniecia (25min)
```

Comment:
Such important variables should be passed as a general, maybe even public reference to this variable, not as just an integer somewhere-in-code. ;)